tests/filters/test_alternative_filter.py

```diff
@@ -179,6 +179,26 @@ def test_filters_scenario_alternatives(self):
             scenario_alternatives = db_map.query(db_map.scenario_alternative_sq).all()
             self.assertEqual(len(scenario_alternatives), 0)
 
+    def test_filters_entity_by_another_alternative_than_its_entity_alternative(self):
+        with DatabaseMapping("sqlite://", create=True) as db_map:
+            self._assert_success(db_map.add_entity_class_item(name="Cat", active_by_default=True))
+            self._assert_success(db_map.add_entity_item(name="Felix", entity_class_name="Cat"))
+            self._assert_success(db_map.add_alternative_item(name="Other"))
+            self._assert_success(
+                db_map.add_entity_alternative_item(
+                    entity_class_name="Cat",
+                    entity_byname=("Felix",),
+                    alternative_name="Base",
+                    active=True,
+                )
+            )
```

Comment:
All `add_xxx_item()` calls could be simplified with the "new" API:

```python
            db_map.add_entity_class(name="Cat", active_by_default=True)
            db_map.add_entity(name="Felix", entity_class_name="Cat")
            db_map.add_alternative(name="Other")
            db_map.add_entity_alternative(
                entity_class_name="Cat",
                entity_byname=("Felix",),
                alternative_name="Base",
                active=True,
            )
```

- No need for `_assert_success()` as the new methods raise errors
- No need to repeat `_item` at every call

This is just a 'heads-up' and 'for your information' etc... I'm not proposing to change the code here.

Reply:
Great, thank you @soininen - will take it into account for the following. Very nice improvement.